src/control/server/server.go

```diff
@@ -79,11 +79,10 @@ func hostname() string {
 	return hn
 }
 
-// Start is the entry point for a daos_server instance.
-func Start(log *logging.LeveledLogger, cfg *config.Server) error {
+func checkConfig(log *logging.LeveledLogger, cfg *config.Server) (*system.FaultDomain, error) {
```

Comment:
Hmm. What I was trying to convey during our conversation was that a function named "checkSomething" implies that you're performing validation on something. This function is doing a little validation but it's also changing a lot of state (setting environment variables, writing out a copy of the configuration, getting the fault domain, etc). 

At a minimum, this could probably use a better name. It would probably be best to further decompose this into smaller functions with distinct areas of responsibility. This would make writing focused unit tests easier.

Reply:
Agreed